op-chain-ops/cmd/op-version-check/main.go

```diff
@@ -0,0 +1,191 @@
+
+	// If no chain IDs are specified, check all chains
+	if len(chainIDs) == 0 {
+		chainIDs = maps.Keys(superchain.OPChains)
+	}
+	slices.Sort(chainIDs)
+
+	targets := make([]*superchain.ChainConfig, 0)
+	for _, chainConfig := range superchain.OPChains {
+		if chainConfig.Superchain == superchainName && slices.Contains(chainIDs, chainConfig.ChainID) {
+			targets = append(targets, chainConfig)
+		}
+	}
+
+	slices.SortFunc(targets, func(i, j *superchain.ChainConfig) int {
+		return int(i.ChainID) - int(j.ChainID)
+	})
+
+	output := []ChainVersionCheck{}
+
+	for _, chainConfig := range targets {
+		clients, err := clients.NewClients(ctx.String("l1-rpc-url"), chainConfig.PublicRPC)
+		if err != nil {
+			return fmt.Errorf("cannot create RPC clients: %w", err)
+		}
+		// The L1Client is required
+		if clients.L1Client == nil {
+			return errors.New("cannot create L1 client")
+		}
+
+		l1ChainID, err := clients.L1Client.ChainID(ctx.Context)
+		if err != nil {
+			return fmt.Errorf("cannot fetch L1 chain ID: %w", err)
+		}
+
+		// The L2Client is not required, but double check the chain id matches if possible
+		if clients.L2Client != nil {
+			l2ChainID, err := clients.L2Client.ChainID(ctx.Context)
+			if err != nil {
+				return fmt.Errorf("cannot fetch L2 chain ID: %w", err)
+			}
+			if chainConfig.ChainID != l2ChainID.Uint64() {
+				return fmt.Errorf("mismatched chain IDs: %d != %d", chainConfig.ChainID, l2ChainID)
+			}
+		}
+
+		log.Info(chainConfig.Name, "l1-chain-id", l1ChainID, "l2-chain-id", chainConfig.ChainID)
+
+		log.Info("Detecting on chain contracts")
+		// Tracking the individual addresses can be deprecated once the system is upgraded
+		// to the new contracts where the system config has a reference to each address.
+		addresses, ok := superchain.Addresses[chainConfig.ChainID]
+		if !ok {
+			return fmt.Errorf("no addresses for chain ID %d", chainConfig.ChainID)
+		}
+
+		versions, err := upgrades.GetContractVersions(ctx.Context, addresses, chainConfig, clients.L1Client)
+		if err != nil {
+			return fmt.Errorf("error getting contract versions: %w", err)
+		}
+
+		output = append(output, ChainVersionCheck{Name: chainConfig.Name, ChainID: chainConfig.ChainID, ContractVersions: versions})
+	}
+
+	// Write contract versions to disk or stdout
+	if outfile := ctx.Path("outfile"); outfile != "" {
+		if err := writeJSON(outfile, output); err != nil {
+			return err
+		}
+	} else {
+		data, err := json.MarshalIndent(output, "", "  ")
+		if err != nil {
+			return err
+		}
+		fmt.Println(string(data))
+	}
+	return nil
+}
+
+// toSuperchainName turns a base layer chain id into a superchain
+// network name.
+func toSuperchainName(chainID uint64) (string, error) {
+	if chainID == 1 {
+		return "mainnet", nil
+	}
+	if chainID == 5 {
+		return "goerli", nil
+	}
+	if chainID == 11155111 {
+		return "sepolia", nil
+	}
+	return "", fmt.Errorf("unsupported chain ID %d", chainID)
+}
+
+func writeJSON(outfile string, input interface{}) error {
```

Comment:
Do you mind looking in `op-service` to see if there is a JSON utility there for writing a file to disk? I believe there is a JSON package, we could begin to deduplicate this write json function that has been already copied into many places

Reply:
ended up using a function from `op-node/cmd/genesis`!